src/streamed/state.ts

```diff
@@ -165,135 +167,258 @@ export class AuctionState {
-        this.lastSolution = { submitter, burntFees }
+        this.applySolutionSubmission(ev.returnValues)
+        this.lastSolution = ev.returnValues
         break
       case "TokenListing":
-        this.addToken(parseInt(data.id), data.token)
+        this.applyTokenListing(ev.returnValues)
         break
       case "Trade":
         this.lastSolution = undefined
-        this.updateBalance(
-          data.owner,
-          parseInt(data.sellToken),
-          amount => amount - BigInt(data.executedSellAmount),
-        )
-        this.updateOrderRemainingAmount(
-          data.owner,
-          parseInt(data.orderId),
-          amount => amount - BigInt(data.executedSellAmount),
-        )
-        this.updateBalance(
-          data.owner,
-          parseInt(data.buyToken),
-          amount => amount + BigInt(data.executedBuyAmount),
-        )
+        this.applyTrade(ev.returnValues)
         break
       case "TradeReversion":
         if (this.lastSolution !== undefined) {
-          const { submitter, burntFees } = this.lastSolution
-          this.updateBalance(submitter, 0, amount => amount - burntFees)
+          this.applySolutionReversion(this.lastSolution)
           this.lastSolution = undefined
         }
-        this.updateBalance(
-          data.owner,
-          parseInt(data.sellToken),
-          amount => amount + BigInt(data.executedSellAmount),
-        )
-        this.updateOrderRemainingAmount(
-          data.owner,
-          parseInt(data.orderId),
-          amount => amount + BigInt(data.executedSellAmount),
-        )
-        this.updateBalance(
-          data.owner,
-          parseInt(data.buyToken),
-          amount => amount - BigInt(data.executedBuyAmount),
-        )
+        this.applyTradeReversion(ev.returnValues)
         break
       case "WithdrawRequest":
-        this.setPendingWithdrawal(
-          data.user,
-          data.token,
-          parseInt(data.batchId),
-          BigInt(data.amount),
-        )
+        this.applyWithdrawRequest(ev.returnValues)
         break
       case "Withdraw":
-        const newBalance = this.updateBalance(
-          data.user,
-          data.token,
-          amount => amount - BigInt(data.amount),
-        )
-        if (this.options.strict) {
-          assert(
-            newBalance >= BigInt(0),
-            `overdrew user ${data.user} token ${data.token} balance`,
-          )
-        }
-        this.clearPendingWithdrawal(data.user, data.token)
+        this.applyWithdraw(ev.returnValues)
         break
       default:
         throw new UnhandledEventError(ev)
       }
     }
-    /* eslint-enable no-case-declarations */
+  }
+
+  /**
+   * Applies a deposit event to the auction state.
+   */
+  private applyDeposit(
+    { user, token, amount: depositAmount }: Event<BatchExchange, "Deposit">,
+  ) {
+    this.updateBalance(user, token, amount => amount + BigInt(depositAmount))
```

Comment:
This code was moved from here:
https://github.com/gnosis/dex-contracts/pull/725/files#diff-4b1723270e0b82c809346fdae9c8d383L179-L183